buildscript/generate-package-infos.gradle

```diff
@@ -0,0 +1,27 @@
+import java.util.regex.Matcher
+
+def generatePackageInfos = tasks.register('generatePackageInfos', Task) {
+    doLast {
+        fileTree('src/main/java').each { javaFile ->
+            def packageInfoFile = new File(javaFile.parent, 'package-info.java')
+            if (!packageInfoFile.exists()) {
+                def pkgName = javaFile.toString().replaceAll(Matcher.quoteReplacement(File.separator), '/')
+                pkgName = pkgName.substring(pkgName.indexOf('net/neoforged/'), pkgName.lastIndexOf('/'))
+                pkgName = pkgName.replaceAll('/', '.')
+
+                def pkgInfoText = """
+                    |@FieldsAreNonnullByDefault
+                    |@MethodsReturnNonnullByDefault
+                    |@ParametersAreNonnullByDefault
+                    |package $pkgName;
+                    |
+                    |import javax.annotation.ParametersAreNonnullByDefault;
+                    |import net.minecraft.FieldsAreNonnullByDefault;
+                    |import net.minecraft.MethodsReturnNonnullByDefault;
+                """.stripMargin().trim()
+
+                packageInfoFile.text = pkgInfoText
+            }
+        }
+    }
+}
```

Comment:
I suggest making this an actual task class with action and not a default task with doLast

Reply:
That can be saved for someone with better gradle/groovy knowledge, I'm already burnt out from getting to this point.